src/components/TokenInput.tsx

```diff
@@ -52,46 +63,85 @@ function TokenInput({
       periodIndex === -1 || e.target.value.length - 1 - periodIndex <= decimals
     if (isValidInput && isValidPrecision) {
       // don't allow input longer than the token allows
-      onChange(e.target.value)
+      const re = /^[0-9]*[.,]?[0-9]*$/
```

Comment:
can you name this const to something that describes what the regex is doing? Something like `readableDecimalNumberRegex` or something.

Reply:
Could you also make this a const and export it from `constants/index`? That would make it easy to reuse in other components and write a couple tests for, since regex is easy to get wrong

Reply:
That would be very useful!